lib/name.mli

```diff
@@ -27,11 +27,20 @@ open Cstruct
 type label
 
 (** Domain name, as a list of strings. *)
-type domain_name = string list
+type domain_name
 
 (** Domain name map *)
 module Map: Map.S with type key = domain_name
 
+(** Create null domain name *)
+val empty: domain_name
+
+(** Render a {! domain_name} to a {! string} list. *)
+val domain_name_to_string_list : domain_name -> string list
```

Comment:
now called to_string_list, i agree with preference for [ ] to {! } but I'll leave for another PR as this is consistent throughout atm